src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const currentState = { ...state };
+
+  const actionHandlers = {
+    addProperties: (action) => {
+      Object.assign(currentState, action.extraData);
+    },
+    removeProperties: (action) => {
+      for (const key of action.keysToRemove) {
+        delete currentState[key];
+      }
+    },
+    clear: () => {
+      for (const key in currentState) {
+        delete currentState[key];
+      }
```

Comment:
if it is a copy, here you can simply overwrite the value with an empty object
```suggestion
      currentState = {};
```